Formula/a/apify-cli.rb

```diff
@@ -0,0 +1,27 @@
+require "language/node"
+
+class ApifyCli < Formula
+  desc "Apify command-line interface"
+  homepage "https://docs.apify.com/cli"
+  url "https://registry.npmjs.org/apify-cli/-/apify-cli-0.18.0.tgz"
+  sha256 "28d4e8b032992560a4f842a44e8d107693b1a031560325b90967245c4ed7b628"
+  license "Apache-2.0"
+
+  depends_on "node@20"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    # We have to replace the shebang in the main executable from "/usr/bin/env node" to point to the keg-only node@16
+    inreplace "#{libexec}/lib/node_modules/apify-cli/src/bin/run", "/usr/bin/env node",
+                                                                   "#{Formula["node@20"].opt_bin}/node"
```

Comment:
This shouldn't be needed anymore.

```suggestion
```

Reply:
So does this mean that if I depend on `node` instead of `node@xx`, Node will then be installed globally, not keg-only?

Reply:
Right. Only the versioned `node`s are keg-only; the unversioned `node` is linked to `$HOMEBREW_PREFIX/bin`.

Reply:
That's a rather nasty side effect for dependencies.

Anyway, I think I still have to do the shebang replacement, because the `node_modules` in `/opt/homebrew/Cellar/apify-cli/0.18.0/libexec/lib` will be installed and compiled using `node` from Homebrew (Node 20 in this case), but if someone would have a different `node` in their `$PATH`, `/usr/bin/env node` would point to that rather than the `node` from Homebrew, and the formula's `node_modules` would be executed using a different version of Node than they were installed with, possibly leading to errors like [this](https://stackoverflow.com/questions/46384591/node-was-compiled-against-a-different-node-js-version-using-node-module-versio):

> Error: The module '/xxx/yyy/zzz'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 93. This version of Node.js requires
NODE_MODULE_VERSION 115. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).

I will just adjust it so that the shebang replacement is done to the unversioned `node` rather than the versioned one.

Reply:
Then you can probably do something like this:

https://github.com/Homebrew/homebrew-core/blob/c1f2c5a1eafaef057e9882def42b218825b7756b/Formula/c/cloudflare-wrangler2.rb#L28

The second argument is the file / list of files to rewrite shebang.

Reply:
Cool, I didn't know `detected_node_shebang` is available, I was writing this formula a few weeks ago and it wasn't there then.

Reply:
Yes, it's a very new thing. It was added two weeks ago in https://github.com/Homebrew/brew/pull/15861.